### PR TITLE
CI: macos-11 [skip azp][skip circle]

### PR DESCRIPTION
**Pull request by andyfaff**

### CODE REVIEW

File: tools/wheels/gfortran_utils.sh [hunk 1]
```diff
@@ -107,47 +105,68 @@ function get_gf_lib_for_suf {
...
+            sudo tar -xvf gfortran-darwin-${arch}-${type}.tar.gz
+            sudo rm gfortran-darwin-${arch}-${type}.tar.gz
         popd
-        export FC_ARM64="$(find /opt/gfortran-darwin-arm64/bin -name "*-gfortran")"
-        export FC_LOC=/opt/gfortran-darwin-arm64/bin
-        local libgfortran="$(find /opt/gfortran-darwin-arm64/lib -name libgfortran.dylib)"
+	if [[ "${type}" == "native" ]]; then
+	    # Link these into /usr/local so that there's no need to add rpath or -L
+	    for f in libgfortran.dylib libgfortran.5.dylib libgcc_s.1.dylib libgcc_s.1.1.dylib libquadmath.dylib libquadmath.0.dylib; do
+                ln -sf /opt/gfortran-darwin-${arch}-${type}/lib/$f /usr/local/lib/$f
```
Comment (isuruf) [inline tools/wheels/gfortran_utils.sh:150]: Until there's a new openblas with this compiler built, you have to add

```suggestion
                ln -sf /opt/gfortran-darwin-${arch}-${type}/lib/$f /usr/local/lib/$f
                mkdir -p /usr/local/gfortran/lib
                ln -sf /opt/gfortran-darwin-${arch}-${type}/lib/$f /usr/local/gfortran/lib/$f
```

Comment (andyfaff) [reply tools/wheels/gfortran_utils.sh]: THanks @isuruf for your help on this. I tried to set `DYLD_LIBRARY_PATH=/usr/local/lib`. If that doesn't work I'll do this.

**Comment by isuruf** [comment]
The new error is an issue with the toolchain. I'll update a new build tomorrow morning.

**Comment by andyfaff** [comment]
@isuruf, the [delocate step still fails](https://github.com/andyfaff/scipy/actions/runs/3408241498/jobs/5668647863#step:5:6409):
```
  ERROR:delocate.libsana:
  /Users/idf2/miniforge/envs/gfortran-darwin-x86_64-native/lib/libgcc_s.1.1.dylib not found:
    Needed by: /opt/gfortran-darwin-x86_64-native/lib/libgcc_s.1.dylib
  ERROR:delocate.libsana:
  /Users/idf2/miniforge/envs/gfortran-darwin-x86_64-native/lib/libgcc_s.1.1.dylib not found:
    Needed by: /opt/gfortran-darwin-x86_64-native/lib/libgcc_s.1.dylib
  ERROR:delocate.libsana:/Users/idf2/miniforge/envs/gfortran-darwin-x86_64-native/lib/libgcc_s.1.1.dylib not found, requested by /opt/gfortran-darwin-x86_64-native/lib/libgcc_s.1.dylib
```

I'm still getting the other @rpath errors, along the lines of:
```
  ERROR:delocate.libsana:
  @rpath/libgfortran.5.dylib not found:
    Needed by: /private/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/tmp__4m2a38/wheel/scipy/optimize/_cobyla.cpython-310-darwin.so
    Search path:
```

**Comment by andyfaff** [comment]
@isuruf, I think there's an issue with the gfortran [cross compiler](https://github.com/isuruf/gcc/releases/download/releases%2Fgcc-11.3.0/gfortran-darwin-arm64-cross.tar.gz) from that script as well.

In a different job I cross compile scipy for cp38-macosx_arm64. i.e. the *host* is `x86_64` and the target is `arm64`. 
I install the cross compiler with `install_arm64_cross_gfortran`. I then try and compile a test program, but the executable has the wrong architecture:

https://github.com/andyfaff/scipy/actions/runs/3410387310/jobs/5673291146#step:6:1059

When I supply the `-arch arm64` flag the compiler complains: `gfortran: error: this compiler does not support arm64`

There is output further on in that run that indicates another problem:
```
-----
Compiler stderr:
/opt/gfortran-darwin-arm64-cross/bin/../libexec/gcc/x86_64-apple-darwin13.4.0/11.3.0/cc1: line 18: /Users/idf2/miniforge/envs/gfortran-darwin-arm64-cross/libexec/gcc/x86_64-apple-darwin13.4.0/11.3.0/cc1.bin: No such file or directory

-----
```

**Comment by isuruf** [comment]
@andyfaff, can you try with https://github.com/MacPython/gfortran-install/pull/11/commits/b665c16e188be1ea551642d9fc678413db052eec ?

**Comment by andyfaff** [comment]
I'm not sure if there's still an issue with the cross compiler. `gfortran-darwin-arm64-cross/lib` contains `libquadmath.dylib`, but `libgfortran.5.dylib` is found in `gfortran-darwin-arm64-cross 2/lib/gcc/arm64-apple-darwin20.0.0/11.3.0`.
Should all the dylibs be located in the same `lib` directory?

**Comment by isuruf** [comment]
Yes, I'll fix that and upload a new build.

**Comment by isuruf** [comment]
Can you try https://github.com/MacPython/gfortran-install/pull/11/commits/d45403b708f751074ef64d19eebe58a30dca3851?

**Comment by andyfaff** [comment]
@isuruf the arm-cross compiler from https://github.com/MacPython/gfortran-install/commit/d45403b708f751074ef64d19eebe58a30dca3851 seems to work in 
https://github.com/andyfaff/scipy/pull/38/commits/d20a1908717f06350e7ef3e14eb71e067e589c7b. I can't comment on whether there are other issues, but the build+delocate works.

**Pull request closed without merging**